### PR TITLE
[DOC] Change desc to descriptor

### PR DESCRIPTION
**Pull request by outoftardis**
# Description

Addresses [this comment](https://github.com/oneapi-src/oneDAL/pull/1522#discussion_r599601793) in #1522

### CODE REVIEW

File: docs/source/api/algorithms/clustering/kmeans-init.rst [hunk 1]
```diff
@@ -75,7 +75,7 @@ Operation
               kmeans_init::compute_result compute(const Descriptor& desc, \
                                          const kmeans_init::compute_input& input)
 
-   :tparam desc: K-Means algorithm descriptor :expr:`kmeans_init::desc`
+   :tparam desc: K-Means algorithm descriptor :expr:`kmeans_init::descriptor`
    :tparam input: Input data for the computing operation
```
Comment (rlnx) [inline docs/source/api/algorithms/clustering/kmeans-init.rst:79]: This is actually `param`, but not `tparam`

Comment (outoftardis) [reply docs/source/api/algorithms/clustering/kmeans-init.rst:79]: only here or everywhere?

File: docs/source/api/algorithms/clustering/kmeans-init.rst [hunk 2]
```diff
@@ -75,8 +75,8 @@ Operation
               kmeans_init::compute_result compute(const Descriptor& desc, \
                                          const kmeans_init::compute_input& input)
 
-   :tparam desc: K-Means algorithm descriptor :expr:`kmeans_init::desc`
-   :tparam input: Input data for the computing operation
+   :param descriptor: K-Means algorithm descriptor :expr:`kmeans_init::descriptor`
```
Comment (rlnx) [inline docs/source/api/algorithms/clustering/kmeans-init.rst:78]:
```suggestion
   :param desc: K-Means algorithm descriptor :expr:`kmeans_init::descriptor`
```

`desc` is name of the argument of the `compute` function above. `descriptor` is a class name. This change should be applied to all algorithms.

Comment (outoftardis) [reply docs/source/api/algorithms/clustering/kmeans-init.rst]: done

File: docs/source/api/algorithms/kernel-functions/linear-kernel.rst [hunk 3]
```diff
@@ -73,7 +73,7 @@ Operation
               linear_kernel::compute_result compute(const Descriptor& desc, \
                                       const linear_kernel::compute_input& input)
 
-   :tparam Descriptor: Linear Kernel algorithm descriptor :expr:`linear_kernel::desc`.
+   :param desc: Linear Kernel algorithm descriptor :expr:`linear_kernel::descriptor`.
```
Comment (rlnx) [inline docs/source/api/algorithms/kernel-functions/linear-kernel.rst:76]: `input` param is missing

File: docs/source/api/algorithms/kernel-functions/rbf-kernel.rst [hunk 4]
```diff
@@ -73,7 +73,7 @@ Operation
               rbf_kernel::compute_result compute(const Descriptor& desc, \
                                       const rbf_kernel::compute_input& input)
 
-   :tparam Descriptor: RBF Kernel algorithm descriptor :expr:`rbf_kernel::desc`.
+   :param desc: RBF Kernel algorithm descriptor :expr:`rbf_kernel::descriptor`.
```
Comment (rlnx) [inline docs/source/api/algorithms/kernel-functions/rbf-kernel.rst:76]: `input` param is missing

File: docs/source/api/algorithms/svm/svm.rst [hunk 5]
```diff
@@ -77,7 +77,7 @@ Operation
               svm::train_result train(const Descriptor& desc, \
                                       const svm::train_input& input)
 
-   :tparam Descriptor: SVM algorithm descriptor :expr:`svm::desc`.
+   :param desc: SVM algorithm descriptor :expr:`svm::descriptor`.
```
Comment (rlnx) [inline docs/source/api/algorithms/svm/svm.rst:80]: `input` param is missing

File: docs/source/api/algorithms/svm/svm.rst [hunk 6]
```diff
@@ -108,7 +108,7 @@ Operation
               svm::infer_result infer(const Descriptor& desc, \
                                       const svm::infer_input& input)
 
-   :tparam Descriptor: SVM algorithm descriptor :expr:`svm::desc`.
+   :param desc: SVM algorithm descriptor :expr:`svm::descriptor`.
```
Comment (rlnx) [inline docs/source/api/algorithms/svm/svm.rst:111]: `input` param is missing

File: docs/source/api/algorithms/ensembles/decision-forest.rst [hunk 7]
```diff
@@ -120,7 +120,7 @@ Operation
               decision_forest::infer_result infer(const Descriptor& desc, \
                                          const decision_forest::infer_input& input)
 
-   :tparam Descriptor: Decision Forest algorithm descriptor :expr:`decision_forest::desc`.
+   :param desc: Decision Forest algorithm descriptor :expr:`decision_forest::descriptor`.
```
Comment (rlnx) [inline docs/source/api/algorithms/ensembles/decision-forest.rst:123]: `input` param is missing

File: docs/source/api/algorithms/ensembles/decision-forest.rst [hunk 8]
```diff
@@ -83,7 +83,7 @@ Operation
               decision_forest::train_result train(const Descriptor& desc, \
                                          const decision_forest::train_input& input)
 
-   :tparam Descriptor: Decision Forest algorithm descriptor :expr:`decision_forest::desc`.
+   :param desc: Decision Forest algorithm descriptor :expr:`decision_forest::descriptor`.
```
Comment (rlnx) [inline docs/source/api/algorithms/ensembles/decision-forest.rst:86]: `input` param is missing

**Review by rlnx — APPROVED** [review]

**Pull request merged**